test/runtests.jl

```diff
@@ -2032,4 +2032,15 @@ end
     f2 = lm(X, filip_data_df.y, dropcollinear = false, method = :qr, wts = ones(length(filip_data_df.y)))
     @test coef(f2) ≈ filip_estimates_df.estimate rtol = 1e-7
     @test stderror(f2) ≈ filip_estimates_df.se rtol = 1e-7
-end
\ No newline at end of file
+end
+
+@testset "Non-finite deviance. Issue 417" begin
+    X_fn = Downloads.download("https://gist.githubusercontent.com/andreasnoack/6068454679cbbc55e0f119ff0d16e92b/raw/de86ddbf66535558b1faedf00bdea59af9a86cb2/X.txt")
+    y_fn = Downloads.download("https://gist.githubusercontent.com/andreasnoack/6068454679cbbc55e0f119ff0d16e92b/raw/acdf8833760b4ea19d7167513ca967b9f7e87c21/y.txt")
```

Comment:
Are these the files included in the issue? You can directly link to them (I know Chris uses this trick sometimes when including e.g. plots in READMEs where he doesn't want the files to be added to the repo):
```suggestion
    X_fn = Downloads.download("https://github.com/JuliaStats/GLM.jl/files/6279630/x.txt")
    y_fn = Downloads.download("https://github.com/JuliaStats/GLM.jl/files/6279632/y.txt")
```